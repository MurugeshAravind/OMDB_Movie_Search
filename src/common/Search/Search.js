import React, {useState} from "react";
import styled from "styled-components";
const WrapperDiv = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
const Input = styled.input.attrs((props) => ({
  type: "text",
  size: props.size || "1em",
  value: props.value || "",
}))`
  border: 2px solid palevioletred;
  margin: ${(props) => props.size};
  padding: ${(props) => props.size};
  flex: 0.5;
`;

const Search = (props) => {
  const {queryValue, page} = props
  const [searchValue, setSearchValue] = useState("");
  const [pageValue, setPageValue] = useState("");
  const handleSearch = (e) => {
    if (e.target.value) {
      setSearchValue(e.target.value)
    } else {
      setSearchValue("")
    }
  };
  searchValue && queryValue(searchValue)
  pageValue && page(pageValue)
  return (
    <WrapperDiv data-testid="search">
      <Input
        placeholder="Enter your movie name"
        onChange={(e) => handleSearch(e)}
        value={searchValue}
      />
    </WrapperDiv>
  );
};
export default Search;
