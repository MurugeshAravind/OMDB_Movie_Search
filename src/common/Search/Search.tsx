import React, {useState} from "react";
import styled from "styled-components";
const WrapperDiv = styled.div`
  display: grid;
  justify-content: center;
  align-items: center;
`;
const Input = styled.input.attrs((props) => ({
  type: "search",
  size: props.size || "1rem",
  value: props.value || "",
}))`
  border: 2px solid palevioletred;
  margin: ${(props) => props.size};
  padding: ${(props) => props.size};
`;

const Search = (props: { queryValue: any; page: any; }) => {
  const {queryValue, page} = props
  const [searchValue, setSearchValue] = useState("");
  const handleSearch = (e: { target: { value: string; }; }) => {
    if (e.target.value) {
      setSearchValue(e.target.value)
      queryValue(e.target.value)
      page(1)
    } else {
      setSearchValue("")
      queryValue("")
    }  
  };
  return (
    <WrapperDiv data-testid="search">
      <Input
        placeholder="Enter your movie name"
        onChange={(e: any) => handleSearch(e)}
        value={searchValue}
      />
    </WrapperDiv>
  );
};
export default Search;
