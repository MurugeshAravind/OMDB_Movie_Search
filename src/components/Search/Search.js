import { useEffect, useState } from "react";
import styled from "styled-components";
import FetchData from "../Api/FetchData";
import * as AppConstants from "../../AppConstants";
const WrapperDiv = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
const Input = styled.input.attrs((props) => ({
  type: "text",
  size: props.size || "1em",
}))`
  border: 2px solid palevioletred;
  margin: ${(props) => props.size};
  padding: ${(props) => props.size};
  flex: 0.5;
`;

const Search = (props) => {
  const [searchValue, setSearchValue] = useState("");
  const [result, setResult] = useState([]);
  const [isLoading, setIsLoading] = useState(false);
  const { searchResult, loaderStatus } = props;
  const handleSearch = (e) => {
    setTimeout(() => {
      const { value } = e.target;
      setSearchValue(value);
    }, 500);
  };
  useEffect(() => {
    if (searchValue) {
      async function fetchMovieList() {
        setIsLoading(true);
        const searchData = await FetchData(
          `${AppConstants.URL}${AppConstants.API_KEY}&s=${searchValue}&type=movie`
        )
          .then((res) => res)
        if (searchData.data.Response === "True") {
          const ids = searchData.data.Search.map((o) => o.imdbID);
          const filtered = searchData.data.Search.filter(
            ({ imdbID }, index) => !ids.includes(imdbID, index + 1)
          );
          setResult(filtered);
          setIsLoading(false);
        } else {
          setResult([]);
          setIsLoading(false);
        }
      }
      fetchMovieList();
    }
  }, [searchValue]);
  useEffect(() => {
    searchResult(result);
  }, [result, searchResult]);
  useEffect(() => {
    loaderStatus(isLoading);
  }, [isLoading, loaderStatus])
  return (
    <WrapperDiv>
      <Input placeholder="Enter your movie name" onChange={handleSearch} />
    </WrapperDiv>
  );
};
export default Search;
