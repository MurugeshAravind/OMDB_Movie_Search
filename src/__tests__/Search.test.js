import React from "react";
import { render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import Router from "react-router-dom";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Search from "../common/Search/Search";
const history = createMemoryHistory();
const search = renderer.create(<Search />).toJSON()
const SearchComponentInstance = () => {
  <Router history={history.location} navigator={history}>
    <Search />
  </Router>;
};
describe("search component", () => {
  test("test the input box in search component", () => {
    render(SearchComponentInstance);
    expect(SearchComponentInstance).toBeTruthy();
    expect(search).toMatchSnapshot();
  })
});
