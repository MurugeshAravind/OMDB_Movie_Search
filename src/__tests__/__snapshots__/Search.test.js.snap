// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search component test the input box in search component 1`] = `
<div
  className="ifSJaK"
>
  <input
    className="iPZaxk"
    onChange={[Function]}
    placeholder="Enter your movie name"
    size="1em"
    type="text"
  />
</div>
`;
